networkx/algorithms/tree/mst.py

```diff
@@ -0,0 +1,323 @@
+    data : bool, optional
+       If True yield the edge data along with the edge.
+
+    Returns
+    -------
+    edges : iterator
+       A generator that produces edges in the minimum spanning tree.
+       The edges are three-tuples (u,v,w) where w is the weight.
+
+    Examples
+    --------
+    >>> from networkx.algorithms import tree
+
+    Find minimum spanning edges by Kruskal's algorithm
+
+    >>> G=nx.cycle_graph(4)
+    >>> G.add_edge(0,3,weight=2) # assign weight 2 to edge 0-3
+    >>> mst=tree.minimum_spanning_edges(G, algorithm='kruskal', data=False) # a generator of MST edges
+    >>> edgelist=list(mst) # make a list of the edges
+    >>> print(sorted(edgelist))
+    [(0, 1), (1, 2), (2, 3)]
+
+    Find minimum spanning edges by Prim's algorithm
+
+    >>> G=nx.cycle_graph(4)
+    >>> G.add_edge(0,3,weight=2) # assign weight 2 to edge 0-3
+    >>> mst=tree.minimum_spanning_edges(G, algorithm='prim', data=False) # a generator of MST edges
+    >>> edgelist=list(mst) # make a list of the edges
+    >>> print(sorted(edgelist))
+    [(0, 1), (1, 2), (2, 3)]
+
+    Notes
+    -----
+
+    If the graph edges do not have a weight attribute a default weight of 1
+    will be used.
+
+    Modified code from David Eppstein, April 2006
+    http://www.ics.uci.edu/~eppstein/PADS/
+    """
+    return spanning_edges(G, minimum=True, algorithm=algorithm,
+                          weight=weight, data=data)
+
+
+@not_implemented_for('directed')
+def maximum_spanning_edges(G, algorithm='kruskal', weight='weight', data=True):
+    """Generate edges in a maximum spanning forest of an undirected
+    weighted graph.
+
+    A maximum spanning tree is a subgraph of the graph (a tree)
+    with the maximum possible sum of edge weights.  A spanning forest is a
+    union of the spanning trees for each connected component of the graph.
+
+    Parameters
+    ----------
+    G : undirected Graph
+       An undirected graph. If `G` is connected, then the algorithm finds a
+       spanning tree. Otherwise, a spanning forest is found.
+
+    algorithm : string
+       The algorithm to use when finding a minimum spanning tree. Valid
+       choices are 'kruskal' or 'prim'. Default is 'kruskal'.
+
+    weight : string
+       Edge data key to use for weight (default 'weight').
+
+    data : bool, optional
+       If True yield the edge data along with the edge.
+
+    Returns
+    -------
+    edges : iterator
+       A generator that produces edges in the maximum spanning tree.
+       The edges are three-tuples (u,v,w) where w is the weight.
+
+    Examples
+    --------
+    >>> from networkx.algorithms import tree
+
+    Find maximum spanning edges by Kruskal's algorithm
+
+    >>> G=nx.cycle_graph(4)
+    >>> G.add_edge(0,3,weight=2) # assign weight 2 to edge 0-3
+    >>> mst=tree.maximum_spanning_edges(G, algorithm='kruskal', data=False) # a generator of MST edges
+    >>> edgelist=list(mst) # make a list of the edges
+    >>> print(sorted(edgelist))
```

Comment:
Sorry to nitpick...but I think these should be cleaned up to match the `*_tree` functions. E.g. there is no need to `print()` and we can probably remove the comments so that they fit in 80 characters/line. I can do this if it doesn't get tackled. Once we have that, let's merge.